haas/model.py

```diff
@@ -40,17 +42,30 @@
 
 db = SQLAlchemy(app)
 
+# Sets up variant type so that postgresql can use BIGINT primary keys
+# while sqlite uses Integer primary keys.
+BigIntegerType = BigInteger().with_variant(
+        sqlite.INTEGER(), 'sqlite')
+
 
 def init_db(uri=None):
     """Start up the DB connection.
 
-    `uri` is the uri to use for the databse. If it is None, the uri from the
+    `uri` is the uri to use for the database. If it is None, the uri from the
     config file will be used.
     """
     if uri is None:
         uri = cfg.get('database', 'uri')
     app.config.update(SQLALCHEMY_DATABASE_URI=uri)
 
+    global BigIntegerType
+
+    # Decide if primary keys should be BIGINTs or Integers
+    if 'postgresql' in uri:
```

Comment:
I have this here so that `BigIntegerType` will instead be a BIGINT if the user has postgresql as a DB back-end. I needed to add this after I learned that sqlite does not willingly accept BIGINTs as primary keys. The sqlite Integer can be 8-bytes max, so this works out.